python/ray/ml/train/impl/function_trainer.py

```diff
@@ -0,0 +1,55 @@
+import abc
+from typing import Dict, Any, Optional
+
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.train import Trainer
+from ray.tune import PlacementGroupFactory
+from ray.util.annotations import DeveloperAPI
+
+
+@DeveloperAPI
+class FunctionTrainer(Trainer):
```

Comment:
copying from slack: @ericl @richardliaw @amogkam  if this is what users interact with, should this be the `Trainer`?

Reply:
The decision here was to:

Try this out for Alpha. Make FunctionTrainer the base Trainer after consensus on FunctionTrainer methods and signatures.